overlord/state/taskrunner.go

```diff
@@ -193,10 +208,7 @@ func (r *TaskRunner) Stop() {
 		tb.Kill(nil)
 	}
 
-	for id, tb := range r.tombs {
-		tb.Wait()
-		delete(r.tombs, id)
-	}
+	r.wait()
```

Comment:
@niemeyer if I understand you mean a flag on the runner set in Stop() and obeyed in Ensure?